internal/auth_test.go

```diff
@@ -0,0 +1,67 @@
+package tfa
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/securecookie"
+)
+
+func TestMakeUserCookie(t *testing.T) {
+	req, _ := http.NewRequest("GET", "http://domain.com", nil)
+	fmt.Println(req)
+
+	config = &Config{
+		CookieHashKey:  cookieHashKey,
+		CookieBlockKey: cookieBlockKey,
+		UserInfoCookie: userInfoCookie,
+	}
+
+	cookie := &http.Cookie{
+		Name:  config.UserInfoCookie,
+		Value: "test_user_info",
+	}
+
+	type args struct {
+		r        *http.Request
+		userInfo string
```

Comment:
Please include more input validations, example: validating input length (max length), which shall avoid buffer overflows.
(all occurrences)

Reply:
thank you. length validation is taken care by `securecookie` library in the Encode() method. we will update the code to handle the error and propagate it up to the api layer.